platform-sdk/swirlds-platform-core/src/test/java/com/swirlds/platform/event/linking/InOrderLinkerTests.java

```diff
@@ -0,0 +1,254 @@
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.swirlds.platform.event.linking;
+
+import static com.swirlds.common.test.fixtures.RandomUtils.getRandomPrintSeed;
+import static com.swirlds.common.test.fixtures.RandomUtils.randomHash;
+import static com.swirlds.platform.event.EventConstants.GENERATION_UNDEFINED;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import com.swirlds.common.crypto.Hash;
+import com.swirlds.common.system.events.BaseEventHashedData;
+import com.swirlds.common.system.events.BaseEventUnhashedData;
+import com.swirlds.common.system.events.EventDescriptor;
+import com.swirlds.platform.event.GossipEvent;
+import com.swirlds.platform.gossip.IntakeEventCounter;
+import com.swirlds.platform.internal.EventImpl;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.function.Consumer;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Tests the {@link InOrderLinker} class
+ */
+class InOrderLinkerTests {
+    /**
+     * How many events have been consumed by the consumer. This value includes to 2 genesis events submitted to the
+     * linker in the {@link #setup()} method.
+     */
+    private AtomicInteger consumedEventCount;
+
+    private AtomicLong exitedIntakePipelineCount;
+    private Random random;
+
+    private InOrderLinker inOrderLinker;
+
+    private GossipEvent genesisSelfParent;
+    private GossipEvent genesisOtherParent;
+
+    /**
+     * The number of events handled in the setup method. Included as a constant for assertion clarity
+     */
+    private final int genesisEventCount = 2;
+
+    /**
+     * Generates a mock event with the given parameters
+     *
+     * @param selfHash              the hash of the event
+     * @param selfGeneration        the generation of the event
+     * @param selfParentHash        the hash of the self parent of the event
+     * @param selfParentGeneration  the generation of the self parent of the event
+     * @param otherParentHash       the hash of the other parent of the event
+     * @param otherParentGeneration the generation of the other parent of the event
+     * @return the mock event
+     */
+    private GossipEvent generateMockEvent(
```

Comment:
there is a GossipEventBuilder which is better suited for this

Reply:
The internal logic of `GossipEventBuilder` is complex and undocumented. Since this component only cares about a small subset of event values, I think it's actually simpler in this case to just mock those few values.

Reply:
Generally, I think its a lot better to build up test tools then having each test do some variation of the same work. If we did that, then GossipEventBuilder would have all the functionality needed for building events for tests. Doesn't have to be for this PR, but I think its a good general rule to follow

Reply:
I agree with that, but what gives me pause here is just Ed's current efforts with modifying events.

With event structure undergoing significant changes, I fear investing to expand a utility to emulate the *current* structure may be wasted effort

Reply:
My PR will absorb your changes and compensate.